src/core_modules/capture-core/components/Pages/ViewEvent/EventDetailsSection/ViewEventDataEntry/viewEventDataEntry.epics.js

```diff
@@ -50,15 +51,17 @@ export const loadViewEventDataEntryEpic = (action$: InputObservable, store: Redu
             const viewEventPage = state.viewEventPage || {};
             return viewEventPage.eventId === eventId && !viewEventPage.showEditEvent;
         }),
-        switchMap(async (action) => {
+        switchMap((action) => {
             const eventContainer = action.payload.eventContainer;
             const orgUnit = action.payload.orgUnit;
             const metadataContainer = getProgramAndStageFromEvent(eventContainer.event);
             if (metadataContainer.error) {
-                return prerequisitesErrorLoadingViewEventDataEntry(metadataContainer.error);
+                return of(prerequisitesErrorLoadingViewEventDataEntry(metadataContainer.error));
```

Comment:
I believe that from the switchMap we should return Observables and not single objects.
This was what flow warning us about here.

Reply:
Returning promises is also valid. Removing the async above caused the problem here, but the solution is good.